File: serverTCC/serverTCC/Models/Agendamento.cs
using System;
using System.Collections.Generic;

namespace serverTCC.Models
{
    public enum DiaSemana
    {
        Domingo,
        Segunda,
        Terca,
        Quarta,
        Quinta,
        Sexta,
        Sabado
    }

    public class Agendamento
    {
        public int Id { get; set; }
        public EscalaTempo EscalaTempo { get; set; }
        public int QtdTempo { get; set; }

        //Até quando repete, se forem nulos, é para sempre
        public DateTime? DataLimite { get; set; } //Caso 1: Repete até uma data específica
        public int? VezesLimite { get; set; } //Caso 2: Repete por uma quantidade de vezes

        //Exclusivo de repetição semanal
        public DiaSemana?[] DiasSemana { get; set; }

        //Exclusivo de repetição mensal
        public int? DiaMes { get; set; } //Caso 1: Repete num dia específico do mes

        public int? SemanaMes { get; set; } //Caso 2: Repete num dia da semana, de uma semana especifica do mes
        public DiaSemana? DiaSemana { get; set; }

    }
}

File: client/www/js/models/agendamento.js
function Agendamento(data) {
    "use strict";
    if (!data) data = {};
    
    this.id = data.id;
    this.escalaTempo = data.escalaTempo;
    this.qtdTempo = data.qtdTempo;
    this.dataLimite = data.dataLimite;
    this.vezesLimite = data.vezesLimite;
    this.diasSemana = data.diasSemana;
    this.diaMes = data.diaMes;
    this.semanaMes = data.semanaMes;
    this.diaSemana = data.diaSemana;
    this.movimentacaoId = data.movimentacaoId;
    if (data.movimentacao)
        this.movimentacao = new Movimentacao(data.movimentacao);
}

localEntities.register(Agendamento, {
    id: {
        pk: true,
        type: 'number'
    },
    escalaTempo: {
        type: 'number'
    },
    qtdTempo: {
        type: 'number'
    },
    dataLimite: {
        type: 'Date'
    },
    vezesLimite: {
        type: 'number'
    },
    diasSemana: {
        type: 'Array:number'
    },
    diaMes: {
        type: 'number'
    },
    semanaMes: {
        type: 'number'
    },
    diaSemana: {
        type: 'number'
    },
    movimentacaoId: {
        fk: true,
        type: 'number'
    },
    movimentacao: {
        relationKey: 'movimentacaoId',
        type: 'Movimentacao'
    }
});